app/models/template.rb

```diff
@@ -5,6 +5,7 @@ class Template < ActiveRecord::Base
   validates_with TemplateLinksValidator
 
   before_validation :set_defaults 
+  after_update :reconcile_published, if: Proc.new { |template| template.published? && template.version.present? && template.version > 0 }
```

Comment:
why those additional checks after published? template.version should always be present and greater than zero

Reply:
just so that it doesn't unecessarily execute the callback for the first time the template is published. Although, I suppose the select statement in the callback would not find any versions to unpublish so it wouldn't be an issue.